src/Persistent/Idiom.hs

```diff
@@ -127,3 +127,10 @@ setTopicPhase tid phase = modifyTopic tid $ topicPhase .~ phase
 deactivateUser :: AUID User -> AUpdate ()
 deactivateUser uid
     = modifyUser uid (userSettings . userSettingsPassword .~ UserPassDeactivated)
+
+getUserViews :: Query [UserView]
+getUserViews = fmap2 makeUserView getAllUsers
+
+findActiveUser :: AUID User -> EQuery User
+findActiveUser =
+    (maybe404 . (^? activeUser) . makeUserView) <=< maybe404 <=< findUser
```

Comment:
Why not define this in `MQuery`?  It seems to me like `maybe404` should only be called inside the application logic, when an `MQuery` is available but an `EQuery` is needed.